megamek/src/megamek/MegaMek.java

```diff
@@ -121,23 +108,22 @@ public static void main(String[] args) {
 
                 String interfaceName = cp.getGuiName();
                 if (interfaceName == null) {
-                    interfaceName = PreferenceManager.getClientPreferences()
-                            .getGUIName();
+                    interfaceName = PreferenceManager.getClientPreferences().getGUIName();
                 }
                 MegaMek.startGUI(interfaceName, restArgs);
             }
 
         } catch (CommandLineParser.ParseException e) {
-            StringBuilder message = new StringBuilder(INCORRECT_ARGUMENTS_MESSAGE)
-                    .append(e.getMessage()).append('\n');
-            message.append(ARGUMENTS_DESCRIPTION_MESSAGE);
-            MegaMek.displayMessageAndExit(message.toString(),
-                                          "main(String[])");
+            String message = INCORRECT_ARGUMENTS_MESSAGE + e.getMessage() + '\n'
+                    + ARGUMENTS_DESCRIPTION_MESSAGE;
+            getLogger().log(MegaMek.class, "main(String[])", LogLevel.INFO, message);
+            TimerSingleton.getInstance().killTimer();
```

Comment:
Is that needed if we're calling System.exit below?

Reply:
Was copying origin functionality, but looking at it again that doesn't make sense to me.

Reply:
It runs as a daemon thread, so not needed, as it will be garbage collected on exit.